R/record_graph.R

```diff
@@ -0,0 +1,37 @@
+
+#' Record a Graph
+#'
+#' @description  This function captures a graph generated by the 'x' object and returns it as a recorded plot.
+#'
+#' @param x An object representing the graph to be recorded.
+#'
+#' @return A recorded plot of the graph generated by 'x'.
+#'
+#' @export
+#'
+#' @examples
+#' # Example 1: Record a scatter plot
+#' data <- data.frame(x = rnorm(100), y = rnorm(100))
+#' plot(data$x, data$y)
+#' recorded_plot <- record_graph(1)
+#'
+#' # Example 2: Record a bar plot
+#' bar_data <- c(A = 10, B = 20, C = 15)
+#' barplot(bar_data)
+#' recorded_plot <- record_graph(2)
+record_graph <- function(x) {
+  # store current device, which could be png
+  d <- dev.cur()
+  # create a new device, to ensure graph is displayed in a device that recordPlot() can capture
+  dev.new()
+  # store the new device, so it can be turned off later
+  d2 <- dev.cur()
+  # Display graph to be captured by recordPlot()
+  x
+  y <- grDevices::recordPlot()
```

Comment:
We are adding in a new package here `grDevices`. Can you edit the DESCRIPTION file to add in the package under imports there?